pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -110,14 +111,29 @@ func (s *Server) initCertController(args *PilotArgs) error {
 //
 // TODO: If the discovery address in mesh.yaml is set to port 15012 (XDS-with-DNS-certs) and the name
 // matches the k8s namespace, failure to start DNS server is a fatal error.
-func (s *Server) initDNSCerts(hostname, namespace string) error {
+func (s *Server) initDNSCerts(hostname, customHostname, namespace string) error {
 	parts := strings.Split(hostname, ".")
 	if len(parts) < 2 {
 		return fmt.Errorf("invalid hostname %s, should contain at least service name and namespace", hostname)
 	}
+
 	// Names in the Istiod cert - support the old service names as well.
 	// The first is the recommended one, also used by Apiserver for webhooks.
 	names := []string{hostname}
+
+	// validate and append custom hostname if there is any
+	if customHostname != "" {
+		partsCustom := strings.Split(customHostname, ".")
+		for _, part := range partsCustom {
+			if !labels.IsDNS1123Label(part) {
```

Comment:
Why would we reject wildcards ? Seems strange, the standard allows them.

Reply:
I removed any validation for this since we didn't validate in 1.5 either